src/test/java/StringManipulationTest.java

```diff
@@ -53,29 +68,70 @@ public void testRemoveNthCharacter2() {
         assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
     }
 
+    // This will test whether the method will correctly remove the characters
+    // in a string with 12 characters by a factor of 2. With a factor of 2 and a
+    // string of length 12, the characters at position 2,4,6,8,10,12 will be removed
+    // and since the maintainSpacing parameter is false, then the characters at those
+    // positions will be deleted.
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+    	manipulatedstring.setString("hello world!");
+    	assertEquals("hlowrd", manipulatedstring.removeNthCharacter(2, false));
     }
-
+    
+    // This will test whether the method will correctly remove the characters
+    // in a string with 12 characters by a factor of 3. With a factor of 3 and a
+    // string of length 12, the characters at position 3,6,9,12 will be removed
+    // and since the maintainSpacing parameter is true, then the characters at those
+    // positions will be replaced with a space character.
+    //
+    // The character in the last position will be replaced with a space which will 
+    // maintain the original string length of 12
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+    	manipulatedstring.setString("hello world!");
+    	assertEquals("he lo wo ld ", manipulatedstring.removeNthCharacter(3, true));
     }
 
+    // This will test the maintainSpacing of the method. The given string has a total of 5 
+    // spaces that are originally in the string. With false passed in for maintainSpacing,
+    // the output string should retain the 5 spaces and remove the characters at positions 3n.
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("0 3  6  9");
+        assertEquals("0     ", manipulatedstring.removeNthCharacter(3, false));
```

Comment:
Good test case.

Would this work if the #n space start is changed? 

I.e. 5n or 10n and so on, but may require a nested test for out of bounds I guess. Hope that somewhat makes sense.